### PR TITLE
Server part

### PR DESCRIPTION
**Pull request by D-moe**
All prior branches up to ApiBlobServe merged together. These changes have already been approved I just need a way to merge them back into master.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/BlobstoreUploadUrlServlet.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  private BlobstoreService blobstoreService;
+  @Override
+  public void init() {
+    blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+  }
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+    String uploadUrl = blobstoreService.createUploadUrl("/data");
+    System.out.println("The upload url is "+uploadUrl);
```
Comment (jmchristen) [inline portfolio/src/main/java/com/google/sps/servlets/BlobstoreUploadUrlServlet.java:42]: Whitespace formatting.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -59,20 +67,90 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
...
+      System.out.println("This is running");
+      String user = entity.getProperty("user").toString();
+      // TODO(morleyd): Change implementation to use a hash to lower time
+      // complexity, need to figure out how this interacts with Gson.
+      // Ensure that duplicate comments (same user and commentData) are not
+      // stored in the datastorage.
+      String comment = entity.getProperty("comment").toString();
+      String blobKey = (String)entity.getProperty("image");
+      // TODO(morleyd): Change implementation to use a hash to lower time
+      // complexity, need to figure out how this interacts with Gson.
```
Comment (jmchristen) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:131]: Duplicate comment.

File: portfolio/src/main/java/com/google/sps/servlets/DeleteData.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+    DatastoreService storeData = DatastoreServiceFactory.getDatastoreService();
+    // TODO(morleyd): See if this code generalizes for >500 entries, think there
+    // might be an upper limit for number of entries deleted at a time
+
+    // TODO(morleyd): See if there is a better way to mass delete values, this
+    // seems like a common task and there is likely some optimization to do it
+    // faster
```
Comment (jmchristen) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteData.java:30]: Duplicate comment.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -53,83 +57,142 @@ function getRandomFile(element, path, fileName, numElems) {
...
           json => {
             for (index in json) {
-              let userName = json[index]['userName'];
-              let commentBody = json[index]['commentBody'];
-              addComment(/* name= */ userName, /* commentBody= */ commentBody);
+              let currentComment = json[index];
+              let userName = currentComment['userName'];
+              let commentBody = currentComment['commentBody'];
+              let blobKey = undefined;
+              // we need to check whether an image Url exists
```
Comment (jmchristen) [inline portfolio/src/main/webapp/script.js:117]: Out of date comment.

File: walkthroughs/week-4-libraries/blobstore/examples/hello-world-fetch/src/main/java/com/google/sps/servlets/BlobstoreUploadUrlServlet.java [hunk 5]
```diff
@@ -23,19 +23,25 @@
...
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+    BlobstoreService blobstoreService =
+        BlobstoreServiceFactory.getBlobstoreService();
     String uploadUrl = blobstoreService.createUploadUrl("/my-form-handler");
 
     response.setContentType("text/html");
+    response.addHeader("Cache-Control",
+                       "no-cache, no-store, must-revalidate, max-age=0");
+   /* response.addHeader("Expires", "0");*/
```
Comment (jmchristen) [inline walkthroughs/week-4-libraries/blobstore/examples/hello-world-fetch/src/main/java/com/google/sps/servlets/BlobstoreUploadUrlServlet.java:44]: Unused code.

**Comment by D-moe** [comment]
I agree on the merging, this way was certainly not ideal.

**Pull request merged**